src/autogluon_dashboard/dashboard.py

```diff
@@ -24,26 +23,103 @@ def upload_to_s3(s3_client: botocore.client, file_name: str, object_name: str, a
+        help="Location of csv file in local filesystem to upload to S3 bucket. Example: sub_folder/file_name.csv",
+    )
+    parser.add_argument(
+        "--all_dataset_csv",
+        type=str,
+        required=True,
+        help="Location of csv file in local filesystem to upload to S3 bucket. Example: sub_folder/file_name.csv",
+    )
+    parser.add_argument(
+        "--per_dataset_s3",
+        type=str,
+        help="Location in S3 bucket to save csv file. Example: sub_folder/file_name.csv",
+        default=CSV_FILES_DIR + "all_data.csv",  # TODO: Add commit ID (if applicable) to S3 file path
+    )
+    parser.add_argument(
+        "--all_dataset_s3",
+        type=str,
+        help="Location in S3 bucket to save csv file. Example: sub_folder/file_name.csv",
+        default=CSV_FILES_DIR + "autogluon.csv",  # TODO: Add commit ID (if applicable) to S3 file path
+    )
+
+    parser.add_argument(
+        "--s3_bucket", type=str, help="Name of S3 bucket that results to aggregate get outputted to", nargs="?"
+    )
+    parser.add_argument("--s3_prefix", type=str, nargs="?", default="")
+    parser.add_argument("--s3_region", type=str, help="S3 Region to deploy the dashboard website", nargs="?")
+
+    args = parser.parse_args()
+    return args
 
 
 def run_dashboard():
-    per_dataset_csv_path = sys.argv[2]
-    aggregated_csv_path = sys.argv[4]
-    per_dataset_s3_loc = "dev_data/all_data.csv"
-    aggregated_s3_loc = "dev_data/autogluon.csv"
-    s3_client = boto3.client("s3")
-    global s3_url
+    args = get_args()
+    # Set variables to corrensponding command line args
+    per_dataset_csv_path = args.per_dataset_csv
+    aggregated_csv_path = args.all_dataset_csv
+    per_dataset_s3_loc = args.per_dataset_s3
+    aggregated_s3_loc = args.all_dataset_s3
+    bucket_name = args.s3_bucket
+    region = args.s3_region
+    prefix = args.s3_prefix
+
+    logger = logging.getLogger("dashboard-logger")
+
+    if not bucket_name:
+        if region:
+            logger.warning(
+                "Cannot specify region if no bucket has been provded. Defaulting to AutoGluon bucket and region (%s)",
+                S3_REGION,
+            )
+        else:
+            logger.warning(
+                "No bucket or region has been provided. Defaulting to AutoGluon bucket and region (%s)",
+                S3_REGION,
+            )
+        region = S3_REGION
+        bucket_name = DEFAULT_BUCKET_NAME  # TODO: Change default bucket name
+    else:
+        if not region:
+            raise ValueError("You must specify a region if you provide a bucket")
+        else:
+            logger.info(f"You have specified Bucket {bucket_name} and Region {region}.")
+
+    # Set s3 public urls to CSVs as environment variables
+    prefix = prefix if prefix.startswith("/") else "/" + prefix
+    prefix = prefix if prefix.endswith("/") else prefix + "/"
+    s3_url = f"https://{bucket_name}{prefix}.s3.{region}.amazonaws.com/"
```

Comment:
I wonder should the correct url be
`f"https://{bucket_name}s3.{region}.amazonaws.com/{prefix}/xxx.html"`?
can you verify?

Reply:
Oh yeah you're right - my bad!

Reply:
Would recommend do an end to end test